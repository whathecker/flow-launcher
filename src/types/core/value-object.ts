export type PriorityTier = "n/a" | "highest" | "high" | "mid" | "low";

export type PriorityMeasure = "n/a" | "yes" | "no"; // TODO: consider to use MIXED type (boolean and null) instead?

export class Priority {
  constructor(
    public tier: PriorityTier,
    public importance: PriorityMeasure,
    public urgency: PriorityMeasure,
  ) {}
}
