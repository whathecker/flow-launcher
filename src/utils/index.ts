export { default as navigationRef } from "./navigationRef";
export { default as taskFilters } from "./taskFilters";
export { default as colorRenderer } from "./colorRenderer";
export { default as labelRenderer } from "./labelRenderer";
