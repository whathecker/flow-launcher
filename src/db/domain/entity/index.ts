import { Priority } from "../value-object";

export type ID = number | string;

export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public id: ID,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
  ) {}
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
    public id?: ID,
  ) {}
}
