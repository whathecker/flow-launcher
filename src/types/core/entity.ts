import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export interface IAddTaskInput {
  title: string;
  description: string;
  goal_id: string;
}

export interface IFetchTasksInput {
  goal_id: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public status: TaskStatus,
    public description?: string,
    public priority?: Priority,
  ) {}
}
