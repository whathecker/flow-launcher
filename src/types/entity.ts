import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[] | [],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
  ) {}
}
