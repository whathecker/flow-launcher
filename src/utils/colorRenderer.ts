import { GoalColor } from "../types/core/entity";
import {
  PriorityTier,
  PriorityBucketColor,
  ColorShadeForFirstGoalColor,
  ColorShadeForSecondGoalColor,
  ColorShadeForThirdGoalColor,
  ColorShadeForFourthGoalColor,
} from "../types/core/value-object";

const assignColorForGoal = (index: number): GoalColor => {
  if (_isFirstColor(index)) return "#E9695B";

  let color;
  switch (_findColorToRender(index)) {
    case 1:
      color = "#84C5E8";
      break;
    case 2:
      color = "#1C88BA";
      break;
    case 3:
      color = "#F0B541";
      break;
    default:
      color = "#E9695B";
      break;
  }

  return color as GoalColor;
};

function _isFirstColor(index: number): boolean {
  if (index === 0 || index % 4 === 0) return true;

  return false;
}

function _findColorToRender(index: number): number {
  return index % 4;
}

const getColorForPrioBucket = (
  goalColor: GoalColor,
  prio: PriorityTier,
): PriorityBucketColor | "#FEFEF8" => {
  let color;
  switch (goalColor) {
    case "#E9695B":
      color = _findColorShadeForFirstColor(prio);
      break;
    case "#84C5E8":
      color = _findColorShadeForSecondColor(prio);
      break;
    case "#1C88BA":
      color = _findColorShadeForThirdColor(prio);
      break;
    case "#F0B541":
      color = _findColorShadeForFourthColor(prio);
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as PriorityBucketColor;
};

function _findColorShadeForFirstColor(
  prio: PriorityTier,
): ColorShadeForFirstGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#C2571A";
      break;
    case "high":
      color = "#DA621E";
      break;
    case "mid":
      color = "#F26D21";
      break;
    case "low":
      color = "#F58B4C";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForFirstGoalColor;
}

function _findColorShadeForSecondColor(
  prio: PriorityTier,
): ColorShadeForSecondGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#107896";
      break;
    case "high":
      color = "#0D9BC3";
      break;
    case "mid":
      color = "#2FAFD3";
      break;
    case "low":
      color = "#65CBE9";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForSecondGoalColor;
}

function _findColorShadeForThirdColor(
  prio: PriorityTier,
): ColorShadeForThirdGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#093145";
      break;
    case "high":
      color = "#0D3D56";
      break;
    case "mid":
      color = "#0D5275";
      break;
    case "low":
      color = "#3C6478";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForThirdGoalColor;
}

function _findColorShadeForFourthColor(
  prio: PriorityTier,
): ColorShadeForFourthGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#BCA136";
      break;
    case "high":
      color = "#D3B53D";
      break;
    case "mid":
      color = "#EBC944";
      break;
    case "low":
      color = "#EFD469";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForFourthGoalColor;
}

export default { assignColorForGoal, getColorForPrioBucket };
