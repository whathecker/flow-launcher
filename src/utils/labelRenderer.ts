import { PriorityTier } from "../types/core/value-object";

const renderPrioBucketLabel = (prio: PriorityTier): string => {
  let label = "";
  switch (prio) {
    case "highest":
      label = "Highest priority";
      break;
    case "high":
      label = "High priority";
      break;
    case "mid":
      label = "Mid priority";
      break;
    case "low":
      label = "Low priority";
      break;
    default:
      // TODO: Add error handling here
      break;
  }
  return label;
};

export default { renderPrioBucketLabel };
