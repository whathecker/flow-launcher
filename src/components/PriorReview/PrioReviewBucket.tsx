import React from "react";
import { SectionList, ListRenderItem } from "react-native";
import { View, Text } from "../Themed";
import { EmptyBucket, TaskReadable } from "./components";
import { Typography, Color } from "../../styles";
import { Task, GoalColor } from "../../types/core/entity";
import { PriorityTier } from "../../types/core/value-object";
import { colorRenderer, labelRenderer } from "../../utils";

type PrioReviewBucketProps = {
  prio: PriorityTier;
  tasks: Task[];
  goalColor: GoalColor;
};

const PrioReviewBucket: React.FC<PrioReviewBucketProps> = ({
  prio,
  tasks,
  goalColor,
}: PrioReviewBucketProps) => {
  const backgroundColor = colorRenderer.getColorForPrioBucket(goalColor, prio);
  const isTasksEmpty = tasks.length === 0;

  const sectionData = [
    {
      title: labelRenderer.renderPrioBucketLabel(prio),
      data: tasks,
    },
  ];

  const renderTaskReadable: ListRenderItem<Task> = ({ item }) => {
    return <TaskReadable title={item.title} />;
  };

  return (
    <View
      style={{
        width: "90%",
        backgroundColor: isTasksEmpty ? "#FEFEF8" : backgroundColor,
        marginTop: 5,
        marginBottom: "3%",
        paddingTop: "4%",
        paddingBottom: "8%",
        borderWidth: 1,
        borderColor: Color.light.defaultBorder,
        borderRadius: 5,
      }}
    >
      {isTasksEmpty ? (
        <EmptyBucket title={labelRenderer.renderPrioBucketLabel(prio)} />
      ) : (
        <View
          style={{
            backgroundColor: backgroundColor,
          }}
        >
          <SectionList
            sections={sectionData}
            renderItem={renderTaskReadable}
            renderSectionHeader={({ section: { title } }) => (
              <Text
                style={{
                  textAlign: "left",
                  ...Typography.h4,
                  fontSize: 19,
                  paddingLeft: "5%",
                  paddingBottom: "5%",
                  color: isTasksEmpty
                    ? Color.light.text
                    : Color.light.whiteText,
                }}
              >
                {title}
              </Text>
            )}
          />
        </View>
      )}
    </View>
  );
};

export default PrioReviewBucket;
