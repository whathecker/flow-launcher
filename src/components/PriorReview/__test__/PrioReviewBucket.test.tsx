import * as React from "react";
import renderer from "react-test-renderer";
import PrioReviewBucket from "../PrioReviewBucket";
import { taskFilters } from "../../../utils";
import { Task } from "../../../types/core/entity";

describe("Test PrioReviewBucket component", () => {
  const tasks: Task[] = [
    {
      _id: "622605f8766c2b210fcc5890",
      description: "",
      goal_id: "61d024a2b4cda2847a224bdf",
      priority: {
        importance: "yes",
        tier: "highest",
        urgency: "yes",
      },
      status: "open",
      title: "Test1",
    },
    {
      _id: "622605fd766c2b210fcc5891",
      description: "",
      goal_id: "61d024a2b4cda2847a224bdf",
      priority: {
        importance: "yes",
        tier: "highest",
        urgency: "yes",
      },
      status: "open",
      title: "Test2",
    },
    {
      _id: "62260602766c2b210fcc5892",
      description: "",
      goal_id: "61d024a2b4cda2847a224bdf",
      priority: {
        importance: "yes",
        tier: "high",
        urgency: "no",
      },
      status: "open",
      title: "test3",
    },
    {
      _id: "62260606766c2b210fcc5893",
      description: "",
      goal_id: "61d024a2b4cda2847a224bdf",
      priority: {
        importance: "yes",
        tier: "high",
        urgency: "no",
      },
      status: "open",
      title: "Test4",
    },
    {
      _id: "62260609766c2b210fcc5894",
      description: "",
      goal_id: "61d024a2b4cda2847a224bdf",
      priority: {
        importance: "no",
        tier: "low",
        urgency: "no",
      },
      status: "open",
      title: "test5",
    },
  ];

  const result = taskFilters.filterByPriorityScheme(tasks);

  it("Should renders PrioReviewBuckets correctly", () => {
    const treeHighest = renderer
      .create(
        <PrioReviewBucket
          prio="highest"
          tasks={result.highest}
          goalColor="#E9695B"
        />,
      )
      .toJSON();

    const treeHigh = renderer
      .create(
        <PrioReviewBucket
          prio="high"
          tasks={result.high}
          goalColor="#84C5E8"
        />,
      )
      .toJSON();

    const treeMid = renderer
      .create(
        <PrioReviewBucket prio="mid" tasks={result.mid} goalColor="#1C88BA" />,
      )
      .toJSON();

    const treeLow = renderer
      .create(
        <PrioReviewBucket prio="low" tasks={result.low} goalColor="#F0B541" />,
      )
      .toJSON();

    expect(treeHighest).toMatchSnapshot();
    expect(treeHigh).toMatchSnapshot();
    expect(treeMid).toMatchSnapshot();
    expect(treeLow).toMatchSnapshot();
  });
});
