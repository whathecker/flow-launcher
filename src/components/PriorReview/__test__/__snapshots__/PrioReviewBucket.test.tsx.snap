// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PrioReviewBucket component Should renders PrioReviewBuckets correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "backgroundColor": "#C2571A",
        "borderColor": "#636262",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": "3%",
        "marginTop": 5,
        "paddingBottom": "8%",
        "paddingTop": "4%",
        "width": "90%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "backgroundColor": "#C2571A",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              Object {
                "_id": "622605f8766c2b210fcc5890",
                "description": "",
                "goal_id": "61d024a2b4cda2847a224bdf",
                "priority": Object {
                  "importance": "yes",
                  "tier": "highest",
                  "urgency": "yes",
                },
                "status": "open",
                "title": "Test1",
              },
              Object {
                "_id": "622605fd766c2b210fcc5891",
                "description": "",
                "goal_id": "61d024a2b4cda2847a224bdf",
                "priority": Object {
                  "importance": "yes",
                  "tier": "highest",
                  "urgency": "yes",
                },
                "status": "open",
                "title": "Test2",
              },
            ],
            "title": "1. Do these before anything else",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "EB-Garamond",
                  "fontSize": 19,
                  "paddingBottom": "5%",
                  "paddingLeft": "5%",
                  "textAlign": "left",
                },
              ]
            }
          >
            1. Do these before anything else
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "alignItems": "center",
                  "borderColor": "#636262",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginLeft": "7.5%",
                  "marginRight": "7.5%",
                  "marginTop": 5,
                  "paddingBottom": 8,
                  "paddingTop": 8,
                  "width": "85%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "Forum-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              Test1
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "alignItems": "center",
                  "borderColor": "#636262",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginLeft": "7.5%",
                  "marginRight": "7.5%",
                  "marginTop": 5,
                  "paddingBottom": 8,
                  "paddingTop": 8,
                  "width": "85%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "Forum-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              Test2
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Test PrioReviewBucket component Should renders PrioReviewBuckets correctly 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "backgroundColor": "#0D9BC3",
        "borderColor": "#636262",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": "3%",
        "marginTop": 5,
        "paddingBottom": "8%",
        "paddingTop": "4%",
        "width": "90%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "backgroundColor": "#0D9BC3",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              Object {
                "_id": "62260602766c2b210fcc5892",
                "description": "",
                "goal_id": "61d024a2b4cda2847a224bdf",
                "priority": Object {
                  "importance": "yes",
                  "tier": "high",
                  "urgency": "no",
                },
                "status": "open",
                "title": "test3",
              },
              Object {
                "_id": "62260606766c2b210fcc5893",
                "description": "",
                "goal_id": "61d024a2b4cda2847a224bdf",
                "priority": Object {
                  "importance": "yes",
                  "tier": "high",
                  "urgency": "no",
                },
                "status": "open",
                "title": "Test4",
              },
            ],
            "title": "2. Sort these important things out",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "EB-Garamond",
                  "fontSize": 19,
                  "paddingBottom": "5%",
                  "paddingLeft": "5%",
                  "textAlign": "left",
                },
              ]
            }
          >
            2. Sort these important things out
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "alignItems": "center",
                  "borderColor": "#636262",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginLeft": "7.5%",
                  "marginRight": "7.5%",
                  "marginTop": 5,
                  "paddingBottom": 8,
                  "paddingTop": 8,
                  "width": "85%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "Forum-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              test3
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "alignItems": "center",
                  "borderColor": "#636262",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginLeft": "7.5%",
                  "marginRight": "7.5%",
                  "marginTop": 5,
                  "paddingBottom": 8,
                  "paddingTop": 8,
                  "width": "85%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "Forum-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              Test4
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Test PrioReviewBucket component Should renders PrioReviewBuckets correctly 3`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "backgroundColor": "#FEFEF8",
        "borderColor": "#636262",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": "3%",
        "marginTop": 5,
        "paddingBottom": "8%",
        "paddingTop": "4%",
        "width": "90%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "color": "#000000",
          "fontFamily": "EB-Garamond",
          "fontSize": 19,
          "paddingLeft": "5%",
        },
      ]
    }
  >
    3. These can wait a bit
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#FEFEF8",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": "6%",
          "paddingTop": "12%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "color": "#848484",
            "fontFamily": "Forum-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      No awaiting tasks
    </Text>
  </View>
</View>
`;

exports[`Test PrioReviewBucket component Should renders PrioReviewBuckets correctly 4`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "backgroundColor": "#EBC944",
        "borderColor": "#636262",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": "3%",
        "marginTop": 5,
        "paddingBottom": "8%",
        "paddingTop": "4%",
        "width": "90%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "backgroundColor": "#EBC944",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              Object {
                "_id": "62260609766c2b210fcc5894",
                "description": "",
                "goal_id": "61d024a2b4cda2847a224bdf",
                "priority": Object {
                  "importance": "no",
                  "tier": "low",
                  "urgency": "no",
                },
                "status": "open",
                "title": "test5",
              },
            ],
            "title": "4. These are for spare time",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "EB-Garamond",
                  "fontSize": 19,
                  "paddingBottom": "5%",
                  "paddingLeft": "5%",
                  "textAlign": "left",
                },
              ]
            }
          >
            4. These are for spare time
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "alignItems": "center",
                  "borderColor": "#636262",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginLeft": "7.5%",
                  "marginRight": "7.5%",
                  "marginTop": 5,
                  "paddingBottom": 8,
                  "paddingTop": 8,
                  "width": "85%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "Forum-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              test5
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
